client/src/components/Join.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import io from 'socket.io-client';
+
+const url = 'http://localhost:8080';
+const socket = io(url);
+
+const Join = ({ setTopLevelState }) => {
+  const [name, setName] = useState('');
+
+  const onChange = (event) => {
+    setName(event.target.value);
+  };
+
+  // Event handlers would need to ..
+  const joinSesson = () => {
+    setTopLevelState('joinName', name);
+    // Communicates to host & DB that this guest is part of this session
+    socket.emit('onJoin', { name, submitted: false });
+    //    Send name (userName) to server to store in DB for that session data
+  };
+
+  return (
+    <form>
+      <label forHtml='login'>Add your name</label>
+      <input type='text' name='login' value={name} onChange={onChange}/>
+      <Link to='/guest-menu' >
+        <button type='submit' onClick={joinSesson}> Join [host]s session </button>
```

Comment:
I noticed during your demo the button was rendering that `[host]s` - should this be some props that you pass down from App? The host should be index 0 in the guests array so you can pass it down along the lines of:  `hostName={this.state.guests[0].guestName}` if you only need the name in this component.

Reply:
I just got rid of this for now. We can add that back in later once we have the code setup to pull session info from the DB.